x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/index/query/H3LatLonGeometry.java

```diff
@@ -117,18 +137,22 @@ public boolean contains(double x, double y) {
 
         @Override
         public PointValues.Relation relate(double minX, double maxX, double minY, double maxY) {
-            GeoArea box = GeoAreaFactory.makeGeoArea(
+            if (minX > this.maxX || maxX < this.minX || maxY < this.minY || minY > this.maxY) {
+                return PointValues.Relation.CELL_OUTSIDE_QUERY;
+            }
+            // h3 edges are fuzzy, therefore to avoid issues when bounding box are around the edges,
+            // we just buffer slightly the bounding box to check if it is inside the h3 bin, otherwise
+            // return crosses.
+            final GeoArea box = GeoAreaFactory.makeGeoArea(
                 PlanetModel.SPHERE,
-                Math.toRadians(maxY),
-                Math.toRadians(minY),
-                Math.toRadians(minX),
-                Math.toRadians(maxX)
+                Math.toRadians(Math.min(GeoUtils.MAX_LAT_INCL, maxY + BBOX_BUFFER_FRACTION)),
```

Comment:
When you calculated the min/max above you used `BBOX_BUFFER_FRACTION * (maxX - minX)`, but here you just use buffer_fraction alone. I think it makes more sense to multiply by the box width, or by the hexagon width.

Reply:
Actually, I just want to increase the edge surface by a fix amount regardless of the size of the bounding box. I have renamed  BBOX_BUFFER_FRACTION to BBOX_EDGE_DELTA and just add a fix value to the edges.